src/lib/state/state.svelte.ts

```diff
@@ -64,11 +67,67 @@ export const globalState = $state<GlobalState>({
     // Indicates if user is using existing approval process, creating one or injected provider
     approvalType: 'existing',
 
+    constructorArguments: {
+      values: {},
+      required: 0
+    },
+
+    deterministic: {
+      isSelected: false,
+      salt: undefined
+    },
+
     // Indicates if deployment is completed.
     completed: false,
   },
 });
 
+export const isValidFormAuthentication = () => globalState.authenticated
+
+const isFormFilledFor = (formInputToCheck: keyof typeof globalState.form) => () => globalState.form[formInputToCheck] !== undefined
+export const isValidFormNetwork = isFormFilledFor("network")
+
+export const isValidConstructorArguments = () => Object.keys(globalState.form.constructorArguments.values).length === globalState.form.constructorArguments.required && Object.values(globalState.form.constructorArguments.values).reduce((checkValueResult: boolean, currentValue) => checkValueResult && (currentValue !== undefined && currentValue !== ""), true)
+
+export const isValidDeterministicConfiguration = () => globalState.form.deterministic.isSelected ? globalState.form.deterministic.salt !== undefined : true
```

Comment:
We should also check that the salt is not empty string.
```suggestion
export const isValidDeterministicConfiguration = () => globalState.form.deterministic.isSelected ? (globalState.form.deterministic.salt !== undefined && globalState.form.deterministic.salt.length > 0) : true;
```